app/controllers/BaseController.java

```diff
@@ -295,7 +299,17 @@ protected JsonNode requestToJSON(Request request) {
         } catch (Throwable ex) {
             throw new InvalidEntityException("Error parsing JSON in request body");
         }
-
         throw new InvalidEntityException("Expected JSON in the request body is missing");
     }
+
+    Metrics getMetrics() {
+        final String requestId = RequestUtils.getRequestId(request());
+        final String cacheKey = Metrics.getCacheKey(requestId);
+        Metrics metrics = (Metrics)Cache.get(cacheKey);
+        if (metrics == null) {
+            metrics = new Metrics(requestId);
+            Cache.set(cacheKey, metrics, METRICS_EXPIRE_SECONDS);
```

Comment:
Wouldn't it be cleaner to do all the metrics work in the interceptor? It could access the user's session using the session as set in the returned header for user-specific stuff like study or userId. It would be nice if this kind of logging were completely orthogonal to the application logic.

I think this would also eliminate the need to cache the metrics object (though that seems to just be in memory, it's a little like a thread local variable, which we were avoiding).